XcodeServerSDKTests/XcodeServerEndpointsTests.swift

```diff
@@ -24,5 +32,291 @@ class XcodeServerEndpointsTests: XCTestCase {
+    }
+    
+    func testEndpointURLCreationForBotsBotRev() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Bots, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue", "endpointURL(.Bots, \(params)) should return \"/api/bots/botValue/revValue\"")
+        }
+    }
+    
+    // MARK: endpointURL(.Integrations)
+    
+    func testEndpointURLCreationForIntegrationsWithoutParams() {
+        let url = self.endpoints?.endpointURL(.Integrations)
+        XCTAssertEqual(url!, "/api/integrations", "endpointURL(.Integrations) should return \"/api/integrations\"")
+    }
+    
+    func testEndpointURLCreationForBotsBotIntegrations() {
+        let paramsArray = [
+            [
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/integrations\"")
+        }
+    }
+    
+    func testEndpointURLCreationForBotsBotRevIntegrations() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/revValue/integrations\"")
+        }
+    }
+    
+    func testEndpointURLCreationForIntegrationsIntegration() {
+        let paramsArray = [
+            [
+                "integration": "integrationValue",
+            ],
+            [
+                "otherKey": "otherValue",
+                "integration": "integrationValue"
+            ],
+            [
+                "rev": "revValue",
+                "otherKey": "otherValue",
+                "integration": "integrationValue"
+            ]
```

Comment:
Okay, if you feel like there's anything that can be removed, can you please do it? In the meantime I'll look at the `rev` stuff. Thanks!